matlab_simulation/04-sensor/scanner_bresenham/collision_pred.m

```diff
@@ -0,0 +1,74 @@
+function [ mode ] = collision_pred( t, X, x_vel, y_vel, map)
+% This function can judge whether the robot will hit the map boundaries or
+% obstacles and returens a strategy (the output "mode") to avoid the
+% collision.
+
+% Encode each hitting situation into a binary code [a,b,c,d], in which 
+% the element "a" represents the coordinate to the left of the predicted robot position,
+% the element "b" represents the coordinate above the predicted robot position,
+% the element "c" represents the coordinate to the right of the predicted robot position,
+% the element "d" represents the coordinate under of the predictied robot position,
+
+% Size of the real map
+[M N] = size(map);
+% Predition of robot positions
+    pred_x = X(1,t-1) + x_vel; % X-axis position at next time step (t)
+    pred_y = X(2,t-1) + y_vel; % Y-axis position at next time step (t)
+
+% Combination of comparision vector (coordinate)
+    if ((pred_x == 1) || (pred_x == N)) % If the robot hit the vertical boundaries
+        if x_vel<0
+          coordinate = [ 1, map( pred_x,pred_y+1), map( pred_x+1, pred_y), map( pred_x, pred_y-1) ]; % Left boundary
+        else x_vel>0
```

Comment:
Should this line and line 30 be elseif rather than else? The code seemed to work which is weird.....